src/index.js

```diff
@@ -57,6 +66,16 @@ function deepForceUpdateStack(instance, shouldUpdate, onUpdate) {
   traverseRenderedChildren(internalInstance, forceUpdateIfPending, onUpdate)
 }
 
+function extractPublicInfoFiber(internalInstance) {
+  const fileName = internalInstance.child &&
+    internalInstance.child._debugSource &&
```

Comment:
Why are you reading filename from the child rather than from the component itself?

Reply:
From what I could see while playing around in Power Editor, if I'm looking for an instance of component Foo, defined in foo.js, It will be Foo's children who will have `_debugSource.fileName === 'foo.js'`. I couldn't test this with stack-based React since it seems it's not that easy to turn that on for Facebook code anymore, but for Fiber, it's working really well in practice.

Reply:
Maybe, I just don't understand why. This doesn't really make sense to me: a child is the child component, not the current component. Maybe you could give an example of component I could look at where children have filenames but parents don't?

Reply:
We talked about this offline. Seems like we can use .displayName, which isn't always available, or use this heuristic, or something else I haven't thought of. Here's what's usually available:

<img width="1126" alt="screen shot 2017-09-21 at 3 23 47 am" src="https://user-images.githubusercontent.com/282817/30691506-c8266d90-9e7c-11e7-9c7a-fbb62db427e8.png">